src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -46,18 +53,27 @@ public NoThreadSocketExecuter() {
    */
   public void wakeup() {
     if(commonSelector != null && commonSelector.isOpen()) {
+      wakeUp = true;
       commonSelector.wakeup();
     }
   }
 
   @Override
   public void setClientOperations(final Client client) {
     ArgumentVerifier.assertNotNull(client, "Client");
-    if(!clients.containsKey(client.getChannel()) || client.isClosed()) {
+    if(!clients.containsKey(client.getChannel())) {
       clients.remove(client.getChannel());
       return;
     }
-    if(client.getChannel().isConnectionPending()) {
+    if(client.isClosed()) {
+      clients.remove(client.getChannel());
+      ListenableFuture<?> lf = schedulerPool.submit(new RemoveFromSelector(commonSelector, client));
+      lf.addListener(new Runnable() {
+        @Override
+        public void run() {
+          PortUtils.closeQuietly(client.getChannel());
```

Comment:
The only reason I mention it is thinking about amount of data to be GC'ed for this frequent but short lived connection case.

In addition to the task, I have been considering potentially doing some lazy construction of the listeners inside futures (to reduce GC if possible, it's still a concept atm).  If we do things like that such optimizations would be useless here.

This seemed like a behavior we could consider optimizing, but I definitely don't want to compromise code complexity in a significant way either.